src/client.cpp

```diff
@@ -394,3 +404,14 @@ Client::download(const std::string &server,
                   callback(res, content_type, original_filename, err);
           });
 }
+
+void
+Client::download_user_avatar(
+  const mtx::identifiers::User &user_id,
+  std::function<void(const mtx::responses::Profile &, RequestErr)> callback)
+{
+        get<mtx::responses::Profile>("/client/r0/profile/" + user_id.toString(),
```

Comment:
This is retrieving the whole profile. So better rename this method to `get_profile`.